src/server/scripts/Spells/spell_dk.cpp

```diff
@@ -734,66 +735,66 @@ class spell_dk_corpse_explosion : public SpellScriptLoader
 };
 
 // 49028 - Dancing Rune Weapon
-class spell_dk_dancing_rune_weapon : public SpellScriptLoader
+enum DancingRuneWeapon
 {
-    public:
-        spell_dk_dancing_rune_weapon() : SpellScriptLoader("spell_dk_dancing_rune_weapon") { }
+    DATA_INITIAL_TARGET_GUID    = 1,
 
-        class spell_dk_dancing_rune_weapon_AuraScript : public AuraScript
-        {
-            PrepareAuraScript(spell_dk_dancing_rune_weapon_AuraScript);
+    SPELL_DK_BLOOD_STRIKE       = 45902,
+    SPELL_DK_ICY_TOUCH          = 45477,
+    SPELL_DK_PLAGUE_STRIKE      = 45462,
+    SPELL_DK_HEART_STRIKE       = 55050,
+    SPELL_DK_DEATH_STRIKE       = 49998
+};
 
-            bool Validate(SpellInfo const* /*spellInfo*/) override
-            {
-                if (!sObjectMgr->GetCreatureTemplate(NPC_DK_DANCING_RUNE_WEAPON))
-                    return false;
-                return true;
-            }
+class spell_dk_dancing_rune_weapon : public AuraScript
+{
+    PrepareAuraScript(spell_dk_dancing_rune_weapon);
 
-            // This is a port of the old switch hack in Unit.cpp, it's not correct
-            void HandleProc(AuraEffect const* /*aurEff*/, ProcEventInfo& eventInfo)
-            {
-                PreventDefaultAction();
-                Unit* caster = GetCaster();
-                if (!caster)
-                    return;
+    void HandleTarget(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        Unit* caster = GetCaster();
+        if (!caster)
+            return;
 
-                Unit* drw = nullptr;
-                for (Unit* controlled : caster->m_Controlled)
-                {
-                    if (controlled->GetEntry() == NPC_DK_DANCING_RUNE_WEAPON)
-                    {
-                        drw = controlled;
-                        break;
-                    }
-                }
+        std::list<Creature*> runeWeapons;
+        caster->GetAllMinionsByEntry(runeWeapons, NPC_DK_DANCING_RUNE_WEAPON);
+        for (Creature* temp : runeWeapons)
+            temp->AI()->SetGUID(GetTarget()->GetGUID(), DATA_INITIAL_TARGET_GUID);
+    }
 
-                if (!drw || !drw->GetVictim())
-                    return;
+    bool CheckProc(ProcEventInfo& eventInfo)
+    {
+        if (SpellInfo const* procSpell = eventInfo.GetSpellInfo())
+            if (procSpell->IsRankOf(sSpellMgr->GetSpellInfo(SPELL_DK_BLOOD_STRIKE)) ||
```

Comment:
This is not the most efficient way of doing this. Pretty sure u can play with values on `spell_proc` table and get the same result.

Reply:
And if you stick to this approach, it lacks a few spells. Rune strike is one of them